### PR TITLE
[TIR][BugFix] Avoid simplifying substituted tir.Any expressions for layout transformations

### PR DESCRIPTION
**Pull request by csullivan**
Avoid simplifying substituted tir.Any expressions when applying shape transformation during layout conversion. Add tests to reproduce the exposed bug observed during ingestion of [ssd_mobilenet_v3_small_coco_2020_01_14](http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v3_small_coco_2020_01_14.tar.gz) without specifying an explicit input shape via the TF frontend.

### CODE REVIEW

File: src/tir/ir/data_layout.cc [hunk 1]
```diff
@@ -323,7 +323,12 @@ inline Array<PrimExpr> TransformShape(const Array<PrimExpr>& src_shape,
       if (symbolic_var_set.count(i)) {
         result.push_back(tir::Any());
       } else {
-        result.push_back(ana.Simplify(tir::Substitute(rule, bind_map)));
+        auto sub = tir::Substitute(rule, bind_map);
+        if (sub.as<tir::AnyNode>()) {
+          result.push_back(tir::Any());
+        } else {
+          result.push_back(ana.Simplify(sub));
+        }
```
Comment (tqchen) [inline src/tir/ir/data_layout.cc:331]: Perhaps in this case we should go and fix Simplify for Any in the arith module and also add a testcase there

Comment (csullivan) [reply src/tir/ir/data_layout.cc:331]: Good point, let me get my head around the semantics and first order simplifications.

Comment (tqchen) [reply src/tir/ir/data_layout.cc:331]: Please followup :)

Comment (csullivan) [reply src/tir/ir/data_layout.cc:331]: I ended up avoiding the layout transformation on the dynamic dimension by specifying it from the input which unblocked me, making this a bit lower priority. I could still try to push this through if you prefer but thinking I will close for now and reopen at a later time.

**Pull request closed without merging**